--- perl-5.16.2/win32/makefile.mk.orig	2012-11-27 21:31:21 +0400
+++ perl-5.16.2/win32/makefile.mk	2012-11-27 21:34:26 +0400
@@ -1432,6 +1432,8 @@
 	if exist $(CCHOME)\bin\libgcc_s_1.dll $(XCOPY) $(CCHOME)\bin\libgcc_s_1.dll ..\t\$(NULL)
 	if exist $(CCHOME)\bin\w64gcc_s_sjlj-1.dll $(XCOPY) $(CCHOME)\bin\w64gcc_s_sjlj-1.dll ..\t\$(NULL)
 	if exist $(CCHOME)\bin\libstdc++-6.dll $(XCOPY) $(CCHOME)\bin\libstdc++-6.dll ..\t\$(NULL)
+	if exist $(CCHOME)\bin\libwinpthread-1.dll $(XCOPY) $(CCHOME)\bin\libwinpthread-1.dll ..\t\$(NULL)
+	if exist $(CCHOME)\bin\libgcc_s_seh-1.dll $(XCOPY) $(CCHOME)\bin\libgcc_s_seh-1.dll ..\t\$(NULL)
 .ENDIF
 
 test : test-prep
