--- qt-stable/qtwebkit/Source/WebCore/DerivedSources.pri.orig	2013-04-15 15:23:31 +0300
+++ qt-stable/qtwebkit/Source/WebCore/DerivedSources.pri	2013-04-15 15:23:35 +0300
@@ -725,7 +725,7 @@
 # GENERATOR 1: Generate .h and .cpp from IDLs
 generateBindings.input = IDL_BINDINGS
 generateBindings.script = $$PWD/bindings/scripts/generate-bindings.pl
-generateBindings.commands = $$setEnvironmentVariable(SOURCE_ROOT, $$toSystemPath($$PWD)) && perl -I$$PWD/bindings/scripts $$generateBindings.script \
+generateBindings.commands = SOURCE_ROOT=$$toSystemPath($$PWD) perl -I$$PWD/bindings/scripts $$generateBindings.script \
                             --defines \"$$javascriptFeatureDefines()\" \
                             --generator JS \
                             --include Modules/filesystem \
