diff --git a/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf b/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf
index 75acbc3..2b2a7b1 100644
--- a/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf
+++ b/qtwebkit/Tools/qmake/mkspecs/features/default_pre.prf
@@ -40,11 +40,7 @@
 }
 
 # Try to detect if we're running under Windows cmd.exe or not
-win32* {
-    pwd = $$(PWD) # Normally empty for cmd.exe
-    pwd_starts_with_slash = $$find(pwd, ^/)
-    isEmpty(pwd_starts_with_slash): CONFIG += win_cmd_shell
-}
+win32:isEmpty(MINGW_IN_SHELL): CONFIG += win_cmd_shell
 
 # Default location of generated sources
 GENERATED_SOURCES_DESTDIR = generated
@@ -55,7 +51,7 @@
     ESCAPE = "^"
 } else {
     QUOTE = "\'"
-    DOUBLE_ESCAPED_QUOTE = "\\\'"
+    DOUBLE_ESCAPED_QUOTE = "\\\''"
 }
 EOL = $$escape_expand(\\n)
 EOC = $$escape_expand(\\n\\t)
