diff --git a/qtwebkit/Source/WebCore/WebCore.pri b/qtwebkit/Source/WebCore/WebCore.pri
index 6608faa..6c83126 100644
--- a/qtwebkit/Source/WebCore/WebCore.pri
+++ b/qtwebkit/Source/WebCore/WebCore.pri
@@ -213,7 +213,11 @@ use?(3D_GRAPHICS) {
         win32-g++: {
             # Make sure OpenGL libs are after the webcore lib so MinGW can resolve symbols
             contains(QT_CONFIG, opengles2) {
-                LIBS += $$QMAKE_LIBS_OPENGL_ES2
+                CONFIG(debug, debug|release):contains(QT_CONFIG, angle) {
+                    LIBS += $$QMAKE_LIBS_OPENGL_ES2_DEBUG
+                } else {
+                    LIBS += $$QMAKE_LIBS_OPENGL_ES2
+                }
             } else {
                 LIBS += $$QMAKE_LIBS_OPENGL
             }
