--- a/qtbase/src/angle/src/config.pri	2013-01-31 22:31:19 +0400
+++ b/qtbase/src/angle/src/config.pri	2013-01-31 22:31:27 +0400
@@ -16,7 +16,7 @@
     GNUTOOLS_DIR=$$PWD/../../../../gnuwin32/bin
     exists($$GNUTOOLS_DIR/gperf.exe) {
         # Escape closing parens when expanding the variable, otherwise cmd confuses itself.
-        GNUTOOLS = "(set PATH=$$replace(GNUTOOLS_DIR, [/\\\\], $${QMAKE_DIR_SEP});%PATH:)=^)%)"
+        GNUTOOLS = "(set $$escape_expand(\\\")PATH=$$replace(GNUTOOLS_DIR, [/\\\\], $${QMAKE_DIR_SEP});%PATH%$$escape_expand(\\\"))"
     }
 }
 
