--- qt-creator-2.7.0-src/qtcreator.pro.orig	2013-04-06 18:23:37 +0400
+++ qt-creator-2.7.0-src/qtcreator.pro	2013-04-06 18:23:39 +0400
@@ -22,7 +22,7 @@
 
 qmake_cache = $$targetPath($$IDE_BUILD_TREE/.qmake.cache)
 equals(QMAKE_DIR_SEP, /): {
-    maybe_quote = "\""
+    maybe_quote = ""
     maybe_backslash = "\\"
 }
 system("echo $${maybe_quote}$${LITERAL_HASH} config for qmake$${maybe_quote} > $$qmake_cache")
